### PR TITLE
os/bluestore: introduce legacy statfs and dev size mismatch alerts

### PR DESCRIPTION
**Pull request by ifed01**
Fixes: https://tracker.ceph.com/issues/39146
Relates to: https://tracker.ceph.com/issues/39151

Signed-off-by: Igor Fedotov <ifedotov@suse.com>

- [x] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

### CODE REVIEW

**Comment by ifed01** [comment]
jenkins retest this please

File: src/os/bluestore/BlueStore.cc [hunk 1]
```diff
@@ -8174,7 +8174,16 @@ int BlueStore::statfs(struct store_statfs_t *buf,
...
+
   if (!per_pool_stat_collection) {
+    if (cct->_conf->bluestore_no_per_pool_stats_tolerance != "enforce" &&
+      cct->_conf->bluestore_warn_on_legacy_statfs) {
+      string s("legacy statfs reporting detected, "
+	"suggest to run store repair to get consistent statistic reports");
+      _set_legacy_statfs_alert(s);
+    } else {
+      _clear_legacy_statfs_alert();
+    }
```
Comment (liewegas) [inline src/os/bluestore/BlueStore.cc]: it seems like this should go in mount() or similar, not every pool_statfs() call?

Comment (liewegas) [reply src/os/bluestore/BlueStore.cc]: actually, we should put it in a helper that raises alerts, and re-call the helper from handle_conf_change(), and add the config option on the watched list, so the user can turn off the warnings without restarting osds.

**Comment by liewegas** [comment]
second patch looks good!

**Comment by ifed01** [comment]
@liewegas - updated

**Comment by liewegas** [comment]
cherry-pick https://github.com/liewegas/ceph/commit/wip-ifed-legacy-statfs-alert ?

**Comment by ifed01** [comment]
> cherry-pick [liewegas@wip-ife](https://github.com/liewegas/ceph/commit/wip-ifed-legacy-statfs-alert) ?

@liewegas - done

**Comment by ifed01** [comment]
@liewegas - shouldn't we backport this to Nautilus?

**Comment by liewegas** [comment]
Yes, definitely!  want to prepare a backport branch?

**Comment by ifed01** [comment]
yeah, will do

**Pull request merged**